framework/src/media/StreamBuffer.h

```diff
@@ -0,0 +1,75 @@
+/******************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#ifndef __MEDIA_STREAMBUFFER_H
+#define __MEDIA_STREAMBUFFER_H
+
+#include <memory>
+#include <mutex>
+#include <condition_variable>
+
+struct rb_s;
+
+namespace media {
+namespace stream {
+
+class StreamBufferReader;
+class StreamBufferWriter;
+class BufferObserverInterface;
+
+class StreamBuffer : public std::enable_shared_from_this<StreamBuffer>
+{
+	friend class StreamBufferReader;
+	friend class StreamBufferWriter;
```

Comment:
Why do you use `friend class`?
I mean why do you want to conceal `notifyObserver` ...

Reply:
Actually, I wish StreamBuffer::read/write can only be called by StreamBufferReader/StreamBufferWriter class.

Reply:
StreamBuffer.h is used internally, so I can change it.
change protected to public, then remove `friend class` ...
what's your opinion?

Reply:
I think it would be better to change the `protected` to `public` and remove the `friend class`.

Reply:
I will fix it

Reply:
Fixed